ACE/ace/WIN32_Asynch_IO.cpp

```diff
@@ -168,25 +168,25 @@ ACE_WIN32_Asynch_Operation::open (const ACE_Handler::Proxy_Ptr &handler_proxy,
 int
 ACE_WIN32_Asynch_Operation::cancel (void)
 {
-#if defined (ACE_HAS_WIN32_OVERLAPPED_IO)
+#if defined (ACE_HAS_CANCEL_IO)
   // All I/O operations that are canceled will complete with the error
   // ERROR_OPERATION_ABORTED. All completion notifications for the I/O
   // operations will occur normally.
 
-  // @@ This API returns 0 on failure. So, I am returning -1 in that
-  //    case. Is that right? (Alex).
+#if (_WIN32_WINNT < _WIN32_WINNT_VISTA)
```

Comment:
Recommend to use 0x0600, older SDKs probably don't have the _WIN32_WINNT_VISTA define

``` suggestion
#if (_WIN32_WINNT < 0x0600)
```